### PR TITLE
Adding eslint structure

### PR DESCRIPTION
**Pull request by gabrielreisn**
This PR adds the agreeded rules and plugin as here: https://docs.google.com/spreadsheets/d/1r5PRSQiyLtu6zsqwoYVYHli2Z8s2NT-3M9_sSOPTQTk/edit#gid=0

It contains 2 folders for `flow` and `typescript` rules (this is required since each one has a different parser and eslint only allow it to have a single one per config)

### CODE REVIEW

**Review by brunodahora — APPROVED** [review]

File: flow/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "extends": [
+    "react-app",
+    "plugin:jest/recommended",
+    "eslint-config-recommended",
+    "plugin:flowtype/recommended"
+  ],
+  "plugins": ["flowtype", "jest", "react", "import", "prettier"],
```
Comment (danielnass) [inline flow/.eslintrc.json:8]: @gabrielreisn WDYT to use `react-hooks` and `jsx-a11y` plugins too?

**Review by danielnass — COMMENTED** [review]
@gabrielreisn see comment

File: flow/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  },
+  "homepage": "https://github.com/pipefy/pipefy-eslint-config#readme",
+  "devDependencies": {
+    "babel-eslint": "^10.0.3",
+    "eslint-config-react-app": "^5.0.2",
+    "eslint-config-recommended": "^4.0.0",
+    "eslint-plugin-flowtype": "^4.5.2",
+    "eslint-plugin-import": "^2.18.2",
+    "eslint-plugin-jest": "^23.0.5",
+    "eslint-plugin-jsx-a11y": "^6.2.3",
```
Comment (thiamsantos) [inline flow/package.json:30]: I don't know if I'm missing something, but some of the installed plugins doesn't seem to be used in the config.

**Review by thiamsantos — APPROVED** [review]

File: flow/.eslintrc.json (hunk 1 above)
Comment (gabrielreisn) [reply flow/.eslintrc.json:8]: both are already included on "exteds: "react-app".." https://www.npmjs.com/package/eslint-config-react-app

File: flow/package.json (hunk 2 above)
Comment (gabrielreisn) [reply flow/package.json:30]: jsx-a11y and react-hooks are part of "esling-config-react-app" https://www.npmjs.com/package/eslint-config-react-app

**Review by mdapper — APPROVED** [review]

**Review by danielnass — APPROVED** [review]

**Review by ralfting — APPROVED** [review]

**Pull request merged**